### PR TITLE
Add cors middleware support in JSON-RPC server

### PR DESCRIPTION
**Pull request by 0xpanoramix**
# Description

This PR adds support for CORS by setting the headers's values based on the already supported server's config file.
Fixes EDGE-64

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [X] I have assigned this PR to myself
- [X] I have added at least 1 reviewer
- [ ] I have tested this code
- [X] I have updated the README and other relevant documents (guides...)
- [ ] I have added sufficient documentation both in code, as well as in the READMEs

# Additional comments

For now, the only header you can set is `Access-Control-Allow-Origin`.

### CODE REVIEW

File: jsonrpc/jsonrpc.go [hunk 1]
```diff
@@ -91,6 +96,24 @@ func (j *JSONRPC) setupHTTP() error {
 	return nil
 }
 
+// The middlewareFactory builds a middleware which enables CORS using the provided config.
+func middlewareFactory(config *Config) func(http.Handler) http.Handler {
+	middleware := func(next http.Handler) http.Handler {
```
Comment (zivkovicmilos) [inline jsonrpc/jsonrpc.go]: You can inline this return

Comment (0xpanoramix) [reply jsonrpc/jsonrpc.go]: Fixed in https://github.com/0xPolygon/polygon-edge/pull/368/commits/91a80b1f9db44a76c758f6c7b04c9a5bc9c2c793

File: command/helper/config.go [hunk 2]
```diff
@@ -56,6 +57,11 @@ type TxPool struct {
 	MaxSlots   uint64 `json:"max_slots"`
 }
 
+// Headers defines the HTTP response headers required to enable CORS.
+type Headers struct {
+	AccessControlAllowOrigin []string `json:"access_control_allow_origin"`
```
Comment (Kourin1996) [inline command/helper/config.go]: `AccessControlAllowOrigins` would be more suitable?

**Review by Kourin1996 — COMMENTED** [review]
Looks good to me, I've left a nitpick comment

**Comment by 0xpanoramix** [comment]
I have read the CLA Document and I hereby sign the CLA

File: command/helper/config.go (hunk 2 above)
Comment (0xpanoramix) [reply command/helper/config.go]: Nice catch :100:, fixed in https://github.com/0xPolygon/polygon-edge/pull/368/commits/381d5af44c6de770fdebf50a1d7e90b2af538b53

**Review by Kourin1996 — CHANGES_REQUESTED** [review]
Looks good to me. But I have realized a few things:
(1) Why don't you add flag to set Access-Control-Allow-Origin?
(2) Allow CORS for all domain here https://github.com/0xPolygon/polygon-edge/blob/5698b85ec9216214a5914eeccb9561fd44dc60ae/jsonrpc/jsonrpc.go#L209. You need to remove this.

**Review by zivkovicmilos — APPROVED** [review]
Please resolve the convo Yoshiki left and you're good to go 🙏 

Also, please make sure you resolve any merge conflicts 👍

**Review by 0xAleksaOpacic — APPROVED** [review]
The code looks good. Have we tested this in the browser? Converting curl to the HTTP request.

**Review by Kourin1996 — APPROVED** [review]
Looks good to me. I tested with local React App and it works. Please resolve conflicts

**Pull request merged**